sdks/go/pkg/beam/core/graph/fn.go

```diff
@@ -569,6 +622,188 @@ func validateSideInputsNumUnknown(processFnInputs []funcx.FnParam, method *funcx
 	return nil
 }
 
+// validateSdfMethods validates that all SDF methods are either present or
+// missing in a Fn, and then returns true if they're present and false
+// otherwise. If some are present and some are missing, it returns an error.
+func validateSdfMethodsPresent(fn *Fn) (bool, error) {
+	// Check if first sdf method is present or not, and compare all subsequent
+	// methods to that result. If there's a mismatch, then we only fail after
+	// finishing the loop so we can output all the missing methods.
+	missing := make([]string, 0)
```

Comment:
Prefer declaring empty slices using the var syntax.

var missing []string

Reply:
Done. I forgot that append works on nil slices.